common/unix-peer.c

```diff
@@ -92,6 +96,21 @@ p11_get_upeer_id (int cfd, uid_t *uid, uid_t *gid, pid_t *pid)
 	if (pid)
 		*pid = -1;
 
+#elif defined(HAVE_GETPEERUCRED)
+	/* *Solaris/OpenIndiana */
+	ucred_t *ucred = NULL;
+	(void)ret;  /* pacify compiler */
+
+	if (getpeerucred(cfd, &ucred) == -1)
+		return -1;
+	if (uid && (*uid = ucred_geteuid(ucred)) == -1)
+		return -1;
```

Comment:
"When  successful,  getpeerucred() stores the pointer to a freshly allo‐
cated ucred_t in the memory location pointed to by the  ucred  argument
if  that memory location contains the null pointer. If the memory loca‐
tion is non-null, it will reuse the existing ucred_t."

So, I will allocate on the stack instead so we don't have to worry about this.